app/src/main/java/org/openobservatory/ooniprobe/receiver/TestRunBroadRequestReceiver.java

```diff
@@ -0,0 +1,122 @@
+import org.openobservatory.ooniprobe.common.PreferenceManager;
+import org.openobservatory.ooniprobe.common.service.RunTestService;
+import org.openobservatory.ooniprobe.test.TestAsyncTask;
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite;
+
+import java.util.List;
+
+public class TestRunBroadRequestReceiver extends BroadcastReceiver implements ServiceConnection {
+    private final EventListener listener;
+    private final PreferenceManager preferenceManager;
+    public RunTestService service;
+    private boolean isBound;
+    private Integer runtime;
+
+    public TestRunBroadRequestReceiver(PreferenceManager preferenceManager, EventListener listener) {
+        this.preferenceManager = preferenceManager;
+        this.listener = listener;
+    }
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        String key = intent.getStringExtra("key");
+        String value = intent.getStringExtra("value");
+        switch (key) {
+            case TestAsyncTask.START:
+                listener.onStart(service);
+                break;
+            case TestAsyncTask.RUN:
+                listener.onRun(value);
+                break;
+            case TestAsyncTask.PRG:
+                List<AbstractSuite> previousTestSuits =
+                        service.task.testSuites.subList(0, service.task.testSuites.indexOf(service.task.currentSuite));
+                int previousTestProgress = (int) Stats.of(Lists.transform(
+                        previousTestSuits,
+                        input -> input.getTestList(preferenceManager).length * 100
+                )).sum();
+                int previousTestRuntime = (int) Stats.of(Lists.transform(
+                        previousTestSuits,
+                        input -> input.getRuntime(preferenceManager)
+                )).sum();
+                int prgs = Integer.parseInt(value);
+                int currentTestRuntime = service.task.currentSuite.getRuntime(preferenceManager);
+                int currentTestMax = service.task.currentSuite.getTestList(preferenceManager).length * 100;
+                double timeLeft = runtime - ((((double) prgs) / currentTestMax * currentTestRuntime) + previousTestRuntime);
+
+                listener.onProgress(previousTestProgress + prgs, timeLeft);
+                break;
+            case TestAsyncTask.LOG:
+                listener.onLog(value);
+                break;
+            case TestAsyncTask.ERR:
+                listener.onError(value);
+                break;
+            case TestAsyncTask.URL:
+                listener.onUrl();
+                break;
+            case TestAsyncTask.INT:
+                listener.onInt();
+                break;
+            case TestAsyncTask.END:
+                listener.onEnd(context);
+                break;
+        }
+    }
+
+    @Override
+    public void onServiceConnected(ComponentName cname, IBinder binder) {
+        //Bind the service to this activity
```

Comment:
```suggestion
        //Bind the service to this activity
```

Is this comment still relevant? This class does not seem to be an activity anymore.